Add guide for migration from google analytics (#933)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/tutorials/migrating-from-google-analytics/index.md

---
title: Migrating to Amplitude from Google Analytics
description: A step-by-step guide to migrate to Amplitude from Google Analytics
---

If you're considering migrating from Google Analytics (GA4) to Amplitude for your analytics and user tracking needs, this guide will help you through the process. Both platforms offer powerful features, but transitioning to Amplitude might require some adjustments. This guide outlines the key steps to ensure a smooth migration.

Migrating from the Google Analytics (GA4) SDK to the Amplitude SDK involves adjusting your tracking implementation to align with Amplitude's event structure and capabilities.

!!!info "Looking to get started easily with low-code migration?"

    While completely replacing Google Analytics (GA4) tracking functions with Amplitude tracking functions is our preferred long-term solution for migrating, Amplitude recognizes that this process can be complex. Amplitude offers a [Google Analytics event forwarder](#using-google-analytics-events-forwarder) that can get you started with your migration quickly, with low-code interaction required from you.

### Initializing Amplitude

> Replace the Google Analytics (GA4) initialization calls with Amplitude initialization calls in your application code. If you were tracking an event in Google Analytics (GA4) like this:

Reviewer: ```suggestion
Replace the Google Analytics (GA4) initialization calls with Amplitude initialization calls in your application code. If you are configuring Google Analytics (GA4) like this:
```

```html
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-XXXXXXXXXX');
</script>
```

Then the corresponding Amplitude call is this:

```html

> <script type="text/javascript">

Author: Ohh i like that. I'll add that in a bit. We'll have the snippet soon.

  amplitude.add('YOUR_API_KEY');
</script>
```

> 

Reviewer: Should include link to all installation methods
```suggestion
For NPM and other installation methods click here.
```

Author: Updated!

It's important to initialize the Amplitude SDK in order to start capturing events and user properties. By including this initialization snippet in your application, you are initializing Amplitude. You must replace `'YOUR_API_KEY'` with your actual Amplitude API key, which you can find in your Amplitude account.

### Tracking events

Amplitude's tracking function require different parameters than Google Analytics (GA4). Refer to the code snippets below for the equivalent tracking functions.

Google Analytics (GA4)

```js
gtag('event', 'event_name', {
  param1: 'value1',
  param2: 'value2',
});
```

> Amplitude

Reviewer: Imo should include links to main SDK docs for all of these methods

Author: Updated!

```js
amplitude.track('event_name, {
  param1: 'value1',
  param2: 'value2',
})
```

### Setting user ID

Google Analytics (GA4)

```js
gtag('config', 'GA_MEASUREMENT_ID', {
    user_id: 'USER_ID',
});
```

Amplitude

```js
amplitude.setUserId('USER_ID');

> ```

Reviewer 2: What about using diff views instead like the other migration docs?

Author: I tried that initially but found that it was a bit overwhelming when the entire block is being replaced. IMO the diff blocks are more useful when only a subset of the function call is changed.

);
```

Refer to [Amplitude Browser SDK documentation](../../sdks/browser-2/index.md) for the most up-to-date and accurate information regarding SDK usage.

### Looking for a low-code solution?

We understand that transitioning from your current analytics platform to a new one can sometimes feel complex, especially when following the migration guide provided. If you're unsure about how to proceed or encounter any challenges along the way, Amplitude offers an alternative and low-code migration process to get started easily.

#### Using Google Analytics events forwarder

Amplitude offers a Google Analytics events forwarder which is a low-code solution that works as a plugin to the Amplitude Browser SDK. Once this plugin is installed on your application, it listens for events that Google Analytics (GA4) tracks. For each event sent to Google Analytics (GA4), a corresponding event is sent to Amplitude.

##### Initializing Amplitude with Google Analytics events forwarder

Add Amplitude SDK with Google Analytics events forwarder snippet right before your Google Tag snippet. Adding it before ensures that all Google Analytics (GA4) events are forwarded to Amplitude.

```html

> <script type="text/javascript">

Author: TODO: update version after public release
